config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java

```diff
@@ -85,10 +86,18 @@ private boolean isAnnotatedEntity(Policy policy, AnnotatedEntity annotatedEntity
                 AnnotatedEntity annotatedPolicyEntity = policyEntity.getAnnotatedEntity();
                 if (policyEntity.isReusable() || isAnnotatedEntity(policyEntity, annotatedEntity)) {
                     if (annotatedPolicyEntity.getId() != null) {
-                        policyEntity.setId(annotatedPolicyEntity.getId());
+                        if(annotatedPolicyEntity.getId().length() > 0 && IdValidator.isValidGoid(annotatedPolicyEntity.getId())){
```

Comment:
I didn't really expect the validation over GOID or GUID contents. I will leave it your decision.

Reply:
Move all sort of validation logic inside isValidGoid method include null or empty checks.

Reply:
@uppoju , if validation is required, can you add these checks at AnnotationDeserializer class instead of here.

Reply:
@RaghuReddy, we will be able to find the exact error message if it has to be handled in AnnotationDeserializer.